### PR TITLE
GH-40674: [GLib] Don't assume gint64 and int64_t use the same type

### PR DESCRIPTION
**Pull request by kou**
### Rationale for this change

GLib doesn't guarantee that `gint64` and `int64_t` use the same type:

https://docs.gtk.org/glib/types.html#gint64

> Note that on platforms with more than one 64-bit standard integer
> type, gint64 and int64_t are not necessarily implemented by the same
> 64-bit integer type. For example, on a platform where both long and
> long long are 64-bit, it might be the case that one of those types is
> used for gint64 and the other is used for int64_t.

### What changes are included in this PR?

Add explicit casts.

### Are these changes tested?

Yes.

### Are there any user-facing changes?

No.
* GitHub Issue: #40674

### CODE REVIEW

**Comment by kou** [comment]
@github-actions crossbow submit homebrew-cpp

**Comment by kou** [comment]
@github-actions crossbow submit homebrew-cpp

**Comment by kou** [comment]
@github-actions crossbow submit homebrew-cpp

File: dev/tasks/homebrew-formulae/github.macos.yml [hunk 1]
```diff
@@ -28,6 +28,8 @@ jobs:
 
       {{ macros.configure_homebrew_arrow(formula)|indent }}
       - name: Test formula
+        env:
+          HOMEBREW_NO_INSTALL_FROM_API: "1"
```
Comment (pitrou) [inline dev/tasks/homebrew-formulae/github.macos.yml:32]: Is this related? Can you add a comment explaining why this is required/desired?

**Review by pitrou — APPROVED** [review]
+1, just a question

File: dev/tasks/homebrew-formulae/github.macos.yml (hunk 1 above)
Comment (kou) [reply dev/tasks/homebrew-formulae/github.macos.yml:32]: I should have extracted this as a separated task.
It seems that homebrew-cpp task is broken. We assume that it uses our configurations (`*.rb`) in https://github.com/apache/arrow/tree/main/dev/tasks/homebrew-formulae but they aren't used. I think that it's related to a Homebrew 4.0.0 change:

* https://brew.sh/2023/02/16/homebrew-4.0.0/
* https://docs.brew.sh/FAQ#can-i-edit-formulae-myself

I thought that this change fixes this but it seems that this change doesn't work...

I'll open an issue for it and revert this change from this PR.

Comment (kou) [reply dev/tasks/homebrew-formulae/github.macos.yml:32]: #40813
